src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3016,12 +3019,37 @@ void addProperties(ImmutableArray<ParameterSymbol> recordParameters)
                 }
             }
 
+            static bool hidesInheritedMember(Symbol symbol, NamedTypeSymbol type)
+            {
+                while ((type = type.BaseTypeNoUseSiteDiagnostics) is object)
+                {
+                    OverriddenOrHiddenMembersHelpers.FindOverriddenOrHiddenMembersInType(
+                        symbol,
```

Comment:
I was suggesting something different in this check, I think.

What we currently do is lookup in the dictionary for signature conflicts. My idea was that we would instead look for conflicts as defined by the same equality metrics as hiding.

So we wouldn't apply this check just for actual inherited members, but all members -- including the ones declared in source.

Effectively, the rule for a "matching" member is hiding, regardless of where that member came from. The difference would be that some synthesized members should not be synthesized if an inherited member exists (like the syntheiszed properties), while some would not be synthesized only is already declared (like the equality members).

Reply:
Thanks. I'll try the hiding comparison for members declared in the record class in the next PR.